src/AppInstallerCommonCore/Public/winget/Registry.h

```diff
@@ -87,14 +87,27 @@ namespace AppInstaller::Registry
         template <Type T>
         typename details::ValueTypeSpecifics<static_cast<DWORD>(T)>::value_t GetValue() const
         {
-            EnsureType(T);
+            THROW_HR_IF(E_INVALIDARG, !HasCompatibleType(T));

```

Comment:
>THROW_HR_IF(E_INVALIDARG, !HasCompatibleType(T)); [](start = 12, length = 49)

nit (for now): Better to implement by calling TryGetValue and throwing when it returns nullopt to prevent future changes from needing to be made to both code paths.